community/detectors/apache_http_server_cve_2021_41773/src/main/java/com/google/tsunami/plugins/detectors/rce/cve_2021_25646/ApacheHttpServerVulnDetector.java

```diff
@@ -0,0 +1,115 @@
+import java.util.Optional;
+
+/**
+ * A {@link VulnDetector} that detects the CVE-2021-41773 vulnerability.
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "ApacheHttpServerVulnDetector",
+    version = "1.0",
+    description = "This detector checks for Apache HTTP Server 2.4.49 Path traversal and disclosure vulnerability.",
+    author = "threedr3am (qiaoer1320@gmail.com)",
+    bootstrapModule = ApacheHttpServerVulnDetectorBootstrapModule.class
+)
+public class ApacheHttpServerVulnDetector implements VulnDetector {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+
+  @Inject
+  ApacheHttpServerVulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "cgi-bin/.%2e/";
+    try {
+      HttpResponse response = httpClient.send(get(targetUri).withEmptyHeaders().build(),
+          networkService);
+      try {
+        Optional<String> server = response.headers().get("Server");
+        Optional<String> body = response.bodyString();
+        if (server.isPresent() && server.get().contains("Apache/2.4.49") && body.isPresent()
+            && !body.get().contains("You don't have permission to access this resource.")) {
+          return true;
+        }
+      } catch (Throwable t) {
+        logger.atInfo().log("Failed to parse cores response json");
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", targetUri);
+    }
+    return false;
+  }
+
+  public DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE-2021-41773"))
```

Comment:
Please use the UPPER_SNAKE_CASE for this ID, i.e. `CVE_2021_41773`.

Reply:
done.